### PR TITLE
WT-6895 Implement HS cursor update method

### PR DESCRIPTION
**Pull request by hbokhari**

### CODE REVIEW

File: src/cursor/cur_hs.c [hunk 1]
```diff
@@ -658,6 +658,87 @@ __curhs_insert(WT_CURSOR *cursor)
...
+
+    hs_cursor = (WT_CURSOR_HS *)cursor;
+    file_cursor = hs_cursor->file_cursor;
+    cbt = (WT_CURSOR_BTREE *)file_cursor;
+    hs_tombstone = hs_upd = NULL;
+
+    CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, update, CUR2BT(file_cursor));
+
+    /* We are assuming that the caller has already searched and found the key. */
+    WT_ASSERT(session, F_ISSET(file_cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET));
```
Comment (quchenhao) [inline src/cursor/cur_hs.c:687]: I mean we should assert F_ISSET(hs_cursor, WT_HS_CUR_COUNTER_SET) also assert WT_ASSERT(session, F_ISSET(file_cursor, WT_CURSTD_KEY_INT)).

File: src/cursor/cur_hs.c [hunk 2]
```diff
@@ -658,6 +658,87 @@ __curhs_insert(WT_CURSOR *cursor)
...
+
+    /*
+     * Only valid scenario to update the history store is to add the stop timestamp. Any other case
+     * should fail.
+     */
+    WT_ASSERT(session, !WT_TIME_WINDOW_IS_EMPTY(&hs_cursor->time_window));
```
Comment (quchenhao) [inline src/cursor/cur_hs.c:693]: This reminds me we can remove the asserts I added in WT-6753

File: src/cursor/cur_hs.c [hunk 3]
```diff
@@ -658,6 +658,87 @@ __curhs_insert(WT_CURSOR *cursor)
...
+    WT_ASSERT(session, WT_TIME_WINDOW_HAS_STOP(&hs_cursor->time_window));
+
+    /* We are positioned on the newest value for user key. */
```
Comment (quchenhao) [inline src/cursor/cur_hs.c:696]: I think this doesn't guarantee we place on the newest.

File: src/cursor/cur_hs.c (hunk 1 above)
Comment (hbokhari) [reply src/cursor/cur_hs.c:687]: I am asserting on HS cursor flags further down. I'll add a check for  `WT_CURSTD_KEY_INT` here.

File: src/cursor/cur_hs.c [hunk 4]
```diff
@@ -658,6 +658,87 @@ __curhs_insert(WT_CURSOR *cursor)
...
+    hs_upd->durable_ts = hs_cursor->time_window.durable_start_ts;
+    hs_upd->txnid = hs_cursor->time_window.start_txn;
+
+    /* Connect the tombstone to the update. */
+    hs_tombstone->next = hs_upd;
+
+    /* Search the page and insert the updates. */
+    WT_WITH_PAGE_INDEX(session, ret = __wt_hs_row_search(cbt, &file_cursor->key, true));
+    WT_ERR(ret);
+    WT_ERR(__wt_hs_modify(cbt, hs_tombstone));
```
Comment (quchenhao) [inline src/cursor/cur_hs.c:731]: I think we need to handle WT_RESTART as well.

File: src/cursor/cur_hs.c (hunk 3 above)
Comment (hbokhari) [reply src/cursor/cur_hs.c:696]: Yes there is no guarantee. As we discussed that offline, there is no easy way to check if the timestamp for the key was set to `WT_TS_MAX` for searching before `update()`  was called. I'll add a comment here and move this assert up.

File: src/cursor/cur_hs.c (hunk 4 above)
Comment (hbokhari) [reply src/cursor/cur_hs.c]: Fixed.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by bvpvamsikrishna — APPROVED** [review]

**Pull request merged**